packages/compiler/src/extractor/extract.ts

```diff
@@ -151,25 +102,190 @@ export const extractProps = (
+        staticStyles.push([k, v.value]);
+      })
+      .with({ type: "Conditional" }, (v) => {
+        conditionalStylesMap.add(v.expression, k, v);
+      })
+      .with({ type: "Record" }, (v) => {
+        const classified = classifyProps(v.value);
+        classified.conditionalPropsList.forEach((cond) => {
+          conditionalStylesMap.add(cond.expression, k, types.conditional(cond));
+        });
+        staticStyles.push([k, classified.staticProps]);
+      })
+      .exhaustive();
+  }
+  return {
+    staticProps: Object.fromEntries(staticStyles),
+    conditionalPropsList: Array.from(
+      conditionalStylesMap.getMap().entries()
+    ).map(([expression, styles]) => ({
+      expression,
+      whenTrue: Object.fromEntries(
+        styles.map(([propName, value]) => [propName, value.whenTrue])
+      ),
+      whenFalse: Object.fromEntries(
+        styles.map(([propName, value]) => [propName, value.whenFalse])
+      ),
+    })),
+  };
+};
+
+export const extractProps = (
+  componentName: (typeof componentList)[keyof typeof componentList],
+  jsx: JsxOpeningElement | JsxSelfClosingElement,
+  props: Record<string, types.Value>
+): { css: string } | undefined => {
+  const { staticProps, conditionalPropsList } = classifyProps(props);
+
+  const staticStyles = evaluateProps(componentName, jsx, staticProps);
+
+  const conditinoalResult = conditionalPropsList.map((conditionalProps) => {
+    const whenTrue = evaluateProps(
+      componentName,
+      jsx,
+      conditionalProps.whenTrue
+    );
+    const whenFalse = evaluateProps(
+      componentName,
+      jsx,
+      conditionalProps.whenFalse
+    );
+    return {
+      expression: conditionalProps.expression,
+      whenTrue,
+      whenFalse,
+    };
+  });
+
+  const conditionalClassNameExpressions: string[] = conditinoalResult.map(
+    (cond) =>
+      `(${cond.expression}) ? ${JSON.stringify(
+        cond.whenTrue?.generatedClassName ?? ""
+      )} : ${JSON.stringify(cond.whenFalse?.generatedClassName ?? "")}`
+  );
+
+  updateClassNameAttr(
+    jsx,
+    conditionalClassNameExpressions.length === 0
+      ? staticStyles?.generatedClassName
+      : {
+        type: "expression",
+        expression: `\`${staticStyles?.generatedClassName
+            ? staticStyles.generatedClassName + " "
+            : ""
+          }${conditionalClassNameExpressions
+            .map((cond) => "${" + cond + "}")
+            .join(" ")}\``,
+      }
+  );
+
+  return {
+    css: [
+      ...(staticStyles?.css ? [staticStyles?.css] : []),
+      ...conditinoalResult.flatMap((cond) => [
+        cond.whenTrue?.css ?? "",
+        cond.whenFalse?.css ?? "",
+      ]),
+    ].join(" "),
+  };
 };
 
 /**
  * Generates a unique key for props, aiding cache efficiency.
  * Incurs O(n log n) cost due to sorting, but it's acceptable given the
  * expensive nature of StyleGenerator's internals.
  */
-// eslint-disable-next-line @typescript-eslint/no-explicit-any -- FIXME
```

Comment:
This ESLint error is an actual bug. This shows that proper TypeScript typing (basically, eliminating `any`) with type-aware ESLint rules are really powerful to detect bugs 🎉 

The error is the following:
```
error  Invalid type "StaticValue" of template literal expression  @typescript-eslint/restrict-template-expressions
```

This indicates that we are trying to serialize object into string, which is not expected. I tested `main` (`2648f5eb5ac8d8cb1a50dd8c84539e314eb4afaa`) and confirmed that [generateKey](https://github.com/poteboy/kuma-ui/blob/2648f5eb5ac8d8cb1a50dd8c84539e314eb4afaa/packages/compiler/src/extractor/extract.ts#L104-L104) returns something like `_hover:[object Object]|bg:#576ddf|borderRadius:14px|color:white|cursor:pointer|fontWeight:600|p:16px 32px`, which is broken since nested object is stringified as `[object Object]`.

Reply:
I filed https://github.com/poteboy/kuma-ui/issues/246